File: src/main/java/nu/peg/slack/pt/util/CommandParser.java
package nu.peg.slack.pt.util;

import com.google.common.base.CharMatcher;
import com.google.common.base.Strings;

import java.util.ArrayList;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;
import java.util.stream.Collectors;

public final class CommandParser {

    private static final Pattern pattern = Pattern.compile("(\"([\\wäöüÄÖÜ,: ]*)\"|([\\wäöüÄÖÜ,:]*))");
    private static final CharMatcher trimMatcher = CharMatcher.is('"').or(CharMatcher.WHITESPACE);

    public static List<String> getArguments(String text) {
        List<String> arguments = new ArrayList<>();

        Matcher matcher = pattern.matcher(text);
        while (matcher.find()) arguments.add(matcher.group());

        return arguments.stream()
                        .map(trimMatcher::trimFrom)
                        .filter(str -> !Strings.isNullOrEmpty(str))
                        .collect(Collectors.toList());
    }
}

File: src/main/kotlin/nu/peg/slack/pt/util/CommandParser.kt
package nu.peg.slack.pt.util

import com.google.common.base.CharMatcher
import com.google.common.base.Strings

import java.util.ArrayList
import java.util.regex.Matcher
import java.util.regex.Pattern
import java.util.stream.Collectors

object CommandParser {

    private val pattern = Pattern.compile("(\"([\\wäöüÄÖÜ,: ]*)\"|([\\wäöüÄÖÜ,:]*))")
    private val trimMatcher = CharMatcher.`is`('"').or(CharMatcher.WHITESPACE)

    fun getArguments(text: String): List<String> {
        val arguments = ArrayList<String>()

        val matcher = pattern.matcher(text)
        while (matcher.find()) arguments.add(matcher.group())

        return arguments.stream()
                .map<String>(Function<String, String> { trimMatcher.trimFrom(it) })
                .filter { str -> !Strings.isNullOrEmpty(str) }
                .collect<List<String>, Any>(Collectors.toList())
    }
}
